tests/api/v1/endpoints/test_connection_config_endpoints.py

```diff
@@ -416,6 +422,34 @@ def test_get_connection_configs(
         assert response_body["page"] == 1
         assert response_body["size"] == page_size
 
+    def test_search_connections(
+        self,
+        connection_config,
+        read_connection_config,
+        api_client: TestClient,
+        generate_auth_header,
+        url,
+    ):
+        auth_header = generate_auth_header(scopes=[CONNECTION_READ])
+
+        resp = api_client.get(url + "?search=primary", headers=auth_header)
+        assert resp.status_code == 200
+        assert len(resp.json()["items"]) == 1
+        assert "primary" in resp.json()["items"][0]["description"].lower()
+
+        resp = api_client.get(url + "?search=read", headers=auth_header)
+        assert resp.status_code == 200
+        assert len(resp.json()["items"]) == 1
+        assert "read" in resp.json()["items"][0]["description"].lower()
+
+        resp = api_client.get(url + "?search=nonexistent", headers=auth_header)
+        assert resp.status_code == 200
+        assert len(resp.json()["items"]) == 0
+
+        resp = api_client.get(url + "?search=postgres", headers=auth_header)
+        assert resp.status_code == 200
+        assert len(resp.json()["items"]) == 2
```

Comment:
It'd be good to assert there's some consistent ordering to this result

Reply:
will do-